### ISSUE TITLE
sample: // forced to next line

### ISSUE THREAD

**Issue by zimbatm** [nix-community/nixpkgs-fmt#123]
Coming form https://discourse.nixos.org/t/nixpkgs-fmt-beta-0-3-1/3812/2?u=zimbatm

/cc @lilyball

## Input
```nix
{
  meta = (oldAttrs.meta or {}) // {
    platforms = pkgs.rustc.meta.platforms;
  };
}
```

## Output
```nix
{
  meta = (oldAttrs.meta or {})
    // {
         platforms = pkgs.rustc.meta.platforms;
       }
    ;
}

```

## Desired output

see input?

**Comment by zimbatm** [comment]
The main argument for that format was to reduce merge conflicts in these cases:

```nix
{
  buildInputs =  [
    curl openssl
  ] ++ lib.optional stdenv.isDarwin darwin.frameworks.Security;
}
```

Once you start having multiple of those, it's nice if each addition is independent from each-other:

```nix
{
  buildInputs =  [
    curl openssl
  ] 
  ++ lib.optional stdenv.isDarwin darwin.frameworks.Security
  ++ enableFoo foopackage
  ++ enableBar barpackage
  ;
}
```

**Comment by lilyball** [comment]
In the merge conflict case, those examples have a full expression after the operator. In the `// {` case, the only thing after the operator is `{`. I could imagine writing code like

```nix
{
  foo = bar
    // { a = 1; }
    // { b = 2; }
    // { c = 3; }
    ;
}
```

but I don't think I've ever seen code in the wild that looks like this. If nixpkgs-fmt wants to produce code like this, I could understand that, but as long as it's putting the contents of the attrset on a new line from the `{`, then putting `// {` at the end of the line instead of on its own line makes more sense. In the multi-operator case that would look like

```nix
{
  foo = bar // {
    a = 1;
  } // {
    b = 2;
  } // {
    c = 3;
  };
}
```

And of course the `;` can go immediately after the `}` because no matter how many of these we add, it's always terminated with the same line `};`.

**Comment by matklad** [comment]
I think this is fixed by #139. At lest, the original example is not modified by the formatter

**Comment by zimbatm** [comment]
yep
